### ISSUE TITLE
incorrect `Use <$>` in 2.1.6

### ISSUE THREAD

**Issue by ghorn** [ndmitchell/hlint#509]
```haskell
{-# OPTIONS_GHC -Wall #-}
{-# LANGUAGE DeriveFunctor #-}
{-# LANGUAGE DeriveFoldable #-}
{-# LANGUAGE DeriveTraversable #-}

data V3 a = V3 a a a deriving (Functor, Foldable, Traversable)

instance Applicative V3 where
  pure x = V3 x x x
  V3 fx fy fz <*> V3 x y z = V3 (fx x) (fy y) (fz z)

genRandom :: IO Double
genRandom = undefined

-- Foo.hs:22:8: Warning: Use <$>
-- Found:
--   sequenceA (pure genRandom)
-- Why not:
--   pure <$> genRandom
--
hlintBug :: IO (V3 Double)
hlintBug = sequenceA (pure genRandom)
```

Desired behavior is to generate 3 different random doubles. Hlint seems to want to generate one random double and use it three times.

**Comment by ndmitchell** [comment]
That's pretty serious! Thanks for the report - I'll put out a release tonight fixing it, with any luck.

**Comment by ndmitchell** [comment]
CC @yairchu who suggested these hints in https://github.com/ndmitchell/hlint/commit/5c3a9e55460c98eef99d8f150e7fa92a0f3c55ca (my fault for all bugs, just thought you'd be curious, no implied blame!)

The assumption here was that `toList (pure x)` produces a singleton list, which it does for most applicatives, but certainly not all, e.g. your V3 applicative. Solution is to remove 2 of the hints in that patch above, which I've done.

**Comment by ndmitchell** [comment]
I'll make a release in the next day with the fix.

**Comment by ghorn** [comment]
Thank you. The quick turnaround really helps.

**Comment by ndmitchell** [comment]
Just finished the upload - 2.1.8 now out.
